File: dev/export-service-sale-py/ibuki.py
from rx import subject
from rx.subject import Subject
from rx.subject import BehaviorSubject
from rx.operators import filter

subject = Subject()
behaviorsubject = BehaviorSubject(0)

def emit(id, options):
    subject.on_next({'id': id, 'data': options})

def filterOn(id):
    return (subject.pipe(filter(
        lambda d: d['id'] == id
    )))
def hotEmit(id, options):
    behaviorsubject.on_next({'id': id, 'data': options})

def hotFilterOn(id):
    return (behaviorsubject.pipe(filter(
        lambda d: d['id'] == id
    )))


class Ibuki():
    def __init__(self):
        pass
    @staticmethod
    def emit(id, options):
        subject.on_next({'id': id, 'data': options})
    
    @staticmethod
    def filterOn(id):
        return (subject.pipe(filter(
            lambda d: d['id'] == id
        )))

    @staticmethod
    def hotEmit(id, options):
        behaviorsubject.on_next({'id': id, 'data': options})

    @staticmethod
    def hotFilterOn(id):
        return (behaviorsubject.pipe(filter(
        lambda d: d['id'] == id
    )))

File: lab/flutter/adam1/lib/classes/ibuki.dart


import 'package:rxdart/rxdart.dart';

class Ibuki {
  static BehaviorSubject<Map<String, dynamic>> subject = BehaviorSubject<Map<String, dynamic>>();

  static void emit(String id, dynamic data){
    subject.add({"id":id, "data": data});
  }

  static Stream<Map<String, dynamic>> filterOn(String id){
    return(subject.stream.where((d) => d['id'] == id));
  }

  static void debounceEmit(String id, dynamic data){
    subject.add({"id":id, "data": data});
  }

  static Stream<Map<String, dynamic>> debounceFilterOn(String id, {int debouncePeriod = 1000}){
    return(subject.stream.where((d) => d['id'] == id)).debounceTime(const Duration(seconds: 5));
  }

}